### PR TITLE
add string functions

### PR DESCRIPTION
**Pull request by rachel-benh**

### CODE REVIEW

File: gamla/functional.py [hunk 1]
```diff
@@ -660,6 +660,16 @@ def drop(seq):
     return drop
 
 
+def replace_in_text(old:str, new:str):
```
Comment (nitzo) [inline gamla/functional.py:663]: typing is wrong, should be `Text`

File: gamla/functional.py [hunk 2]
```diff
@@ -660,6 +660,16 @@ def drop(seq):
...
+    def replace_in_text(text: str):
+        return text.replace(old, new)
+
```
Comment (nitzo) [inline gamla/functional.py:666]: missing return on the inner function

File: gamla/functional.py [hunk 3]
```diff
@@ -660,6 +660,16 @@ def drop(seq):
     return drop
 
 
+def replace_in_text(old:str, new:str):
+    def replace_in_text(text: str):
```
Comment (nitzo) [inline gamla/functional.py:664]: typing is wrong, add return type

File: gamla/functional.py [hunk 4]
```diff
@@ -660,6 +660,16 @@ def drop(seq):
...
+        return text.replace(old, new)
+
+
+def split_text(sep:str):
```
Comment (nitzo) [inline gamla/functional.py:668]: typing is wrong. should be `Text`

File: gamla/functional.py [hunk 5]
```diff
@@ -660,6 +660,16 @@ def drop(seq):
...
+    def split_text(text: str):
+        return text.split(sep)
+
```
Comment (nitzo) [inline gamla/functional.py:671]: missing return of inner function

File: gamla/functional.py [hunk 6]
```diff
@@ -660,6 +660,16 @@ def drop(seq):
     return drop
 
 
+def replace_in_text(old:str, new:str):
+    def replace_in_text(text: str):
+        return text.replace(old, new)
+
+
+def split_text(sep:str):
+    def split_text(text: str):
```
Comment (nitzo) [inline gamla/functional.py:669]: typing + return type

**Comment by nitzo** [comment]
Please add tests + docstrings

**Review by nitzo — APPROVED** [review]

**Comment by uriva** [comment]
could've been nice to put them in a string namespace

**Pull request merged**